src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -585,12 +585,43 @@
 examples:
   - name: Create role assignment for an assignee.
     text: az role assignment create --assignee sp_name --role a_role
+  - name: Create role assignment for an assignee with description and condition. (Only for 2020-04-01-preview API version and later.)
+    text: >-
+        az role assignment create --role "Owner" --assignee "Jhon.Doe@Contoso.com"
+        --description "Role assignment foo to check on bar"
+        --condition "@Resource[Microsoft.Storage/storageAccounts/blobServices/containers:Name] stringEquals 'foo'"
+        --condition-version "2.0"
   - name: Create a new role assignment for a user, group, or service principal. (autogenerated)
     text: |
         az role assignment create --assignee 00000000-0000-0000-0000-000000000000 --role "Storage Account Key Operator Service Role" --scope $id
     crafted: true
 """
 
+
+helps['role assignment update'] = """
+type: command
+short-summary: Update an existing role assignment for a user, group, or service principal.
+examples:
+  - name: Update a role assignment from a JSON file.
+    text: az role assignment update --role-assignment assignment.json
+  - name: Update a role assignment from a JSON string. (Bash)
+    text: |
+        az role assignment update --role-assignment '{
+            "canDelegate": null,
```

Comment:
Can we add parameters like --can-deleagate so that users can see help messages.

Reply:
Taking JSON as an input is asked by the service team. I had the same suggestion as you to expose explicit arguments but the suggestion was rejected.

Reply:
CLI is yours. You fix it. Be brave. Go ahead.